### DISCUSSION TITLE
Lesson 2. How to get the challenge NFT?

### DISCUSSION THREAD

**Discussion by ppoojjaa** [Cyfrin/foundry-full-course-cu#127] [Q&A] [answered]
In lesson 2's solveChallenge() in " Write contract ", I am supposed to input a number. Is there any challenge or code to get this number? I am unable to find how to get that number.

Any help would be appreciated, 
Thank you

**Comment by atraxsrc** [comment]
Read the the lines word by word, take your time. SOME will click at the end.
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/92285717/73bcb360-9147-435f-a45b-37bde7f8b63c)

**Comment by ShubhSensei** [comment]
The number is 123 as stated in a variable ``` someNumber ```. 
Make sure to read the contracts more carefully next time 👍

**Comment by ShubhSensei** [comment, marked as answer]
@ppoojjaa  Here's the answer. Scan QR > click on write contract (connect wallet)> solveChallenge > enter no. as 123 > then click on write button. 

<img width="557" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/91625601/6a4d0c64-3cef-4397-9e96-8cf12b268a18">

**Reply by cromewar** [reply]
Nice approach, but remember not to spoil the challenges. They are supposed to be difficult to solve 👀

**Reply by ShubhSensei** [reply]
Got it ser 🫡

**Reply by MultiverseMonacoa** [reply]
I appreciate that there was at least some guidance from this guy for the first one from someone for newer peeps... like how would one know that they didn't need to copy/paste this into solidity to try to compile and run instead of going to the write tab on etherscan? Some of these things can be daunting to figure out for the first time and all the pros have a hard time seeing things with a beginners view so they leave out important stepping stones

**Reply by ShubhSensei** [reply]
Yeah, thanx for understanding 😊

**Reply by MisterMadrid** [reply]
Honestly I spent like 2 hours thinking I needed to play around with the contract and yeah.. no dice lol so I'm glad I found this cuz I was getting super frustrated lol

**Reply by ShubhSensei** [reply]
Hope it helps 😁👍🏻
